model/histogram/float_histogram.go

```diff
@@ -321,24 +381,45 @@ func (h *FloatHistogram) Add(other *FloatHistogram) *FloatHistogram {
 	h.PositiveSpans, h.PositiveBuckets = addBuckets(h.Schema, h.ZeroThreshold, false, hPositiveSpans, hPositiveBuckets, otherPositiveSpans, otherPositiveBuckets)
 	h.NegativeSpans, h.NegativeBuckets = addBuckets(h.Schema, h.ZeroThreshold, false, hNegativeSpans, hNegativeBuckets, otherNegativeSpans, otherNegativeBuckets)
 
-	return h
+	h.CustomBounds = nil
+
+	return h, nil
 }
 
 // Sub works like Add but subtracts the other histogram.
-func (h *FloatHistogram) Sub(other *FloatHistogram) *FloatHistogram {
-	otherZeroCount := h.reconcileZeroBuckets(other)
-	h.ZeroCount -= otherZeroCount
+func (h *FloatHistogram) Sub(other *FloatHistogram) (*FloatHistogram, error) {
+	if h.HasCustomBounds() != other.HasCustomBounds() {
+		return nil, ErrHistogramsIncompatibleSchema
+	}
+	if h.HasCustomBounds() && !floatBucketsMatch(h.CustomBounds, other.CustomBounds) {
+		return nil, ErrHistogramsIncompatibleBounds
+	}
+
+	if !h.HasCustomBounds() {
+		otherZeroCount := h.reconcileZeroBuckets(other)
+		h.ZeroCount -= otherZeroCount
+	}
 	h.Count -= other.Count
 	h.Sum -= other.Sum
 
 	var (
-		hPositiveSpans   = h.PositiveSpans
-		hPositiveBuckets = h.PositiveBuckets
-		hNegativeSpans   = h.NegativeSpans
-		hNegativeBuckets = h.NegativeBuckets
-
+		hPositiveSpans       = h.PositiveSpans
+		hPositiveBuckets     = h.PositiveBuckets
 		otherPositiveSpans   = other.PositiveSpans
 		otherPositiveBuckets = other.PositiveBuckets
+	)
+
+	if h.HasCustomBounds() {
+		h.ZeroThreshold = 0
+		h.ZeroCount = 0
```

Comment:
I think it's enough to ensure these in constructors/copy constructors. Here it feels a bit too much and also will be hard to ensure we zero out everything, suggest to just ignore those fields.